src/app.rs

```diff
@@ -10,22 +10,44 @@ use {
         terminal, ExecutableCommand, QueueableCommand,
     },
     notify::{RecommendedWatcher, RecursiveMode, Watcher},
-    std::{
-        env,
-        fs,
-        io::Write,
-        path::PathBuf,
-    },
+    std::{env, fs, io::Write, path::PathBuf, process::Command},
     termimad::{Event, EventSource},
+    cargo_metadata::{Metadata, MetadataCommand}
 };
 
+
+fn find_folders_to_watch(cargo_filepath: &PathBuf) -> Result<Vec<PathBuf>> {
+    let metadata = MetadataCommand::new().manifest_path(cargo_filepath).exec()?;
+
+    let mut folders_to_watch = vec![];
+
+    for item in metadata.packages {
+        if item.source.is_none() {
+            // We're only concerned with items where the source is None
+            for target in item.targets {
+                if target.src_path.ends_with("lib.rs") || target.src_path.ends_with("main.rs") {
```

Comment:
This doesn't seem like the right criteria. You can explicitly rename both of these to be something else. Instead you should look at `crate_type` and `kind`; unfortunately the logic is a little hairy: https://github.com/deadlinks/cargo-deadlinks/blob/master/src/main.rs#L157

Your logic will be slightly different because you don't care about documentation - I'm not sure exactly what this check is for tbh.

Reply:
The logic was quite flawed. For some reason, I was looking into the targets defined within the package, and trying to find the folder off of that.

Switched to directly using `parent()` of `pacakge.manifest_path`